core/domain/classifier_services.py

```diff
@@ -0,0 +1,301 @@
+# coding: utf-8
+#
+# Copyright 2015 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import numpy
```

Comment:
Nit: I don't think numpy is part of the Python standard library, so following the [style guide](https://google-styleguide.googlecode.com/svn/trunk/pyguide.html#Imports_formatting) let's do:

```
    import copy
    import operator

    import numpy
```